pkg/controller/authentication/authentication_controller.go

```diff
@@ -134,6 +176,17 @@ func (r *ReconcileAuthentication) Reconcile(request reconcile.Request) (reconcil
 	r.status.OnCRFound()
 	reqLogger.V(2).Info("Loaded config", "config", authentication)
 
+	// Verify the presence and status of the installation resource.
+	if installationCR == nil {
```

Comment:
I think this will be removed if you follow through with my suggestion to return an error if the installationCR is not found